strapi/strapi
# fix import of @strapi/icons/Puzzle in generated admin plugin

### What does it do?

Fixes the import  of @strapi/icons/Puzzle in generated admin plugin

Modify the import statement to match how the module is exported by `@strapi/icons` and add the module
as a dependency to the relevant `package.json`

### Why is it needed?

The generated admin plugin was building with errors due to the incorrect import

### How to test it?

Create a project using the latest version of Strapi - 4.5.2
Follow the instructions on how to [create a plugin](https://docs.strapi.io/developer-docs/latest/development/plugins-development.html#create-a-plugin)

```
npm run strapi generate
```

and choose `plugin` from the menu. Choose TypeScript as the language. 

Then follow the other steps to install and build. 

You will see errors about the import of `@strapi/icons/Puzzle`

If you view the file `src/plugins/<your-plugin>/admin/src/components/PluginIcon/index.tsx` in an IDE you will see errors about the import. 

The errors can be manually fixed by installing `@strapi/icons` in the plugin directory - `npm install @strapi/icons`
and by changing the line 

```ts
import Puzzle from '@strapi/icons/Puzzle';
```

to 

```ts
import Puzzle from '@strapi/icons';
```

This PR updates the relevant templates with those fixes. 




### Related issue(s)/PR(s)

none
